tests/test_monitor.py

```diff
@@ -0,0 +1,77 @@
+
+    def output(self):
+        return luigi.LocalTarget('dummy.txt')
+
+
+class TestSuccessTask(luigi.Task):
+    num = luigi.Parameter()
+
+    def output(self):
+        return luigi.LocalTarget('words.txt')
+
+    def run(self):
+        # write a dummy list of words to output file
+        words = [
+            'apple',
+            'banana',
+            'grapefruit'
+        ]
+
+        with self.output().open('w') as f:
+            for word in words:
+                f.write('{word}\n'.format(word=word))
+
+
+# This method will be used by the mock to replace requests.get
+def mocked_requests_post(*args, **kwargs):
+    resp = mock.Mock(spec=requests.Response)
+    resp.status_code = 200
+    return resp
+
+
+class TestLuigiMonitor(unittest.TestCase):
+    @mock.patch('requests.post', side_effect=mocked_requests_post)
+    def test_monitor_no_message(self, mock_post):
+        with monitor(slack_url='mock://slack', events=['DEPENDENCY_MISSING']) as m:
+            luigi.run(main_task_cls=TestSuccessTask, local_scheduler=True, cmdline_args=['--num', '1'])
+            self.assertDictEqual(m.recorded_events, {})
+        self.assertEqual(mock_post.call_count, 0, 'Slack webhook called')
+
+    @mock.patch('requests.post', side_effect=mocked_requests_post)
+    def test_monitor_missing_message(self, mock_post):
+        with monitor(slack_url='mock://slack', events=['DEPENDENCY_MISSING']) as m:
+            luigi.run(main_task_cls=TestMissingTask, local_scheduler=True, cmdline_args=[])
+            self.assertDictEqual(m.recorded_events, {'Missing': ['TestMissingTask()']})
+        call_data = mock_post.call_args[1]['data']
+        call_url = mock_post.call_args[0][0]
+        expected = '{"text": "Status report for ' + os.path.basename(inspect.stack()[-1][1]) + '\\n' \
+                                                                                               '*Tasks with missing dependencies:*' \
+                                                                                               '\\n' \
+                                                                                               'TestMissingTask()"}'
+        self.assertEqual(call_data, expected)
+
+    @mock.patch('requests.post', side_effect=mocked_requests_post)
+    def test_monitor_success_message(self, mock_post):
+        with monitor(slack_url='mock://slack') as m:
+            luigi.run(main_task_cls=TestSuccessTask, local_scheduler=True, cmdline_args=['--num', '2'])
+            self.assertDictEqual(m.recorded_events, {'Success': ['TestSuccessTask(num=2)']})
+        call_data = mock_post.call_args[1]['data']
+        call_url = mock_post.call_args[0][0]
+        expected = '{"text": "Status report for ' + os.path.basename(
+            inspect.stack()[-1][1]) + '\\nJob ran successfully!"}'
+        self.assertEqual(call_data, expected)
+
+    def tearDown(self):
+        if os.path.isfile('words.txt'):
+            os.remove('words.txt')
```

Comment:
`words.txt` could be pulled out into a module-level constant to make sure the file created is always the file deleted

Reply:
I actually got rid of the whole `tearDown()` method as I'm using a `MockTarget` from luigi which is in-memory target. so file system is not used for writing at all.